sqlglot/parser.py

```diff
@@ -4341,6 +4341,9 @@ def _parse_assignment(self) -> t.Optional[exp.Expression]:
             )
 
         while self._match_set(self.ASSIGNMENT):
+            if isinstance(this, exp.Column) and len(this.parts) == 1:
```

Comment:
i think it should be a var. i’m surprised we don’t do this already, additionally should we parse union value as a struct?

Reply:
I think the correct representation is to use an identifier, because the same case-sensitivity rules apply here. For example, in duckdb it's case insensitive so quotes won't matter. Other properties specified using this operator also work with quotes too, e.g. I think you can do `"recursive" => true` in unnest.

> additionally should we parse union value as a struct?

They're not equivalent, see https://duckdb.org/docs/sql/data_types/union.

Reply:
Although I thought it wouldn't be correct, quoting DuckDB properties such as `recursive` does work for whatever reason:

```
D select unnest([], recursive := true) as nonquoted, unnest([], "recursive" := true) as quoted;
┌───────────┬────────┐
│ nonquoted │ quoted │
│   int32   │ int32  │
├───────────┴────────┤
│       0 rows       │
└────────────────────┘
```

So parsing the LHS as Identifier seems like the safest bet afaict

Reply:
is this true for all use of := ? not just unions?

Reply:
I believe so, extrapolating based on the recursive unnest example.